extensions/eda/plugins/event_filter/normalize_keys.py

```diff
@@ -1,46 +1,46 @@
-"""normalize_keys.py.
+import logging
+import multiprocessing as mp
+import re
+from typing import Any
 
-    An event filter that changes keys that contain non alpha numeric or
+DOCUMENTATION = r"""
+---
+author:
+  - Doston Toirov (@dtoirov)
+short_description: Change keys that contain non-alpha numeric or underscore to underscores.
+description: |
+  - An event filter that changes keys that contain non alpha numeric or
     underscore to underscores.
-    For instance, the key server-name becomes the new key server_name
-    If there are consecutive non alpa numeric or under score, they would
-    be coalesced into a single underscore
+  - For instance, the key server-name becomes the new key server_name.
+    If there are consecutive non alpha numeric or under score, they would
+    be coalesced into a single underscore.
     For instance the key server.com/&abc becomes server_com_abc
     instead of server_com__abc.
-
-    If there is a existing key with the normalized name, it will get overwritten
-    by default. If you don't want to over write it you can pass in overwrite: False
-    The default value of overwrite is True.
-
-Arguments:
----------
-    * overwrite: Overwrite the values if there is a collision with a new key.
-
-Usage in a rulebook, a filter is usually attached to a source in the rulebook:
-    example1:
-    ....
-    sources:
-      - my_source:
-         ....
-        filters:
-          ansible.eda.normalize_keys:
-
-
-    example2: To prevent over writing of keys
-    ....
-    sources:
-      - my_source:
-         ....
-        filters:
-          ansible.eda.normalize_keys:
-            overwrite: False
-
+  - If there is a existing key with the normalized name, it will get overwritten
+    by default. If you don't want to over write it you can pass in "overwrite: false"
+    The default value of overwrite is true.
+options:
+  overwrite:
+    description:
+      - Overwrite the values if there is a collision with a new key.
+    type: bool
+    default: true
 """
 
-import logging
-import multiprocessing as mp
-import re
-from typing import Any
+EXAMPLES = r"""
+- sources:
+    - my_source:
+        ....
+      filters:
+        ansible.eda.normalize_keys:
+- name: prevent overwriting of keys
```

Comment:
This is confusing. The source has also a name field. Use a newline to distinguish different sources. Better if you remove the "sources" key root. That would be aligned to regular ansible modules, which have the task definition without the `tasks` key root.